cli/azd/cmd/hooks.go

```diff
@@ -0,0 +1,267 @@
+
+func (hra *hooksRunAction) Run(ctx context.Context) (*actions.ActionResult, error) {
+	hookName := hra.args[0]
+
+	// Command title
+	hra.console.MessageUxItem(ctx, &ux.MessageTitle{
+		Title: "Running hooks (azd hooks run)",
+		TitleNote: fmt.Sprintf(
+			"Finding and executing %s hooks for environment %s",
+			output.WithHighLightFormat(hookName),
+			output.WithHighLightFormat(hra.env.GetEnvName()),
+		),
+	})
+
+	// Validate service name
+	if _, ok := hra.projectConfig.Services[hra.flags.service]; hra.flags.service != "" && !ok {
+		return nil, fmt.Errorf("service name '%s' doesn't exist", hra.flags.service)
+	}
+
+	// Project level hooks
+	projectHooksMessage := "Running command hook for project"
+	if err := hra.processHooks(
+		ctx,
+		hra.projectConfig.Path,
+		hookName,
+		projectHooksMessage,
+		hra.projectConfig.Hooks,
+		false,
+	); err != nil {
+		return nil, err
+	}
+
+	// Service level hooks
+	for serviceName, service := range hra.projectConfig.Services {
+		skip := hra.flags.service != "" && serviceName != hra.flags.service
+
+		serviceHookMessage := fmt.Sprintf("Running service hook for %s", serviceName)
+		if err := hra.processHooks(
+			ctx,
+			service.RelativePath,
+			hookName,
+			serviceHookMessage,
+			service.Hooks,
+			skip,
+		); err != nil {
+			return nil, err
+		}
+	}
+
+	return &actions.ActionResult{
+		Message: &actions.ResultMessage{
+			Header: "Your hooks have been run successfully",
+		},
+	}, nil
+}
+
+func (hra *hooksRunAction) processHooks(
+	ctx context.Context,
+	cwd string,
+	hookName string,
+	message string,
+	hooks map[string]*ext.HookConfig,
+	skip bool,
+) error {
+	hra.console.ShowSpinner(ctx, message, input.Step)
+
+	if skip {
+		hra.console.StopSpinner(ctx, message, input.StepSkipped)
+		return nil
+	}
+
+	hook, ok := hooks[hookName]
+	if !ok {
+		hra.console.StopSpinner(ctx, message+noHookFoundMessage, input.StepWarning)
+		return nil
+	}
+
+	hookType, commandName := ext.InferHookType(hookName)
+
+	if err := hra.prepareHook(hookName, hook); err != nil {
+		return err
+	}
+
+	if hook.Interactive {
+		hra.console.StopSpinner(ctx, "", input.StepDone)
+		hra.console.Message(ctx, output.WithBold(output.WithUnderline(message)))
+	}
+
+	err := hra.execHook(ctx, cwd, hookType, commandName, hook)
+	if err != nil {
+		hra.console.StopSpinner(ctx, message, input.StepFailed)
+		return fmt.Errorf("failed running hook %s, %w", hookName, err)
+	}
+
+	hra.console.StopSpinner(ctx, message, input.StepDone)
+	return nil
+}
+
+func (hra *hooksRunAction) execHook(
+	ctx context.Context,
+	cwd string,
+	hookType ext.HookType,
+	commandName string,
+	hook *ext.HookConfig,
+) error {
+	hookName := string(hookType) + commandName
+
+	hooks := map[string]*ext.HookConfig{
+		hookName: hook,
+	}
+
+	hooksManager := ext.NewHooksManager(cwd)
+	hooksRunner := ext.NewHooksRunner(hooksManager, hra.commandRunner, hra.console, cwd, hooks, hra.env)
+	err := hooksRunner.RunHooks(ctx, hookType, commandName)
```

Comment:
Guessing that long term the Quiet stuff will move to become a property of the HooksRunner, perhaps?

Reply:
Yes, we'll come up with something better.